### PR TITLE
Upgrade for compatibility with Toxiproxy 2.0

### PR DESCRIPTION
**Pull request by Sirupsen**
This makes the Ruby gem compatible with v2.0.0rc1 (https://github.com/Shopify/toxiproxy/pull/83).

@byroot @fw42 @xthexder

### CODE REVIEW

**Comment by byroot** [comment]
LGTM.

File: README.md [hunk 1]
```diff
@@ -50,12 +50,13 @@ If you want to simulate that your cache server is slow at incoming network
 upstream:
```
Comment (fw42) [inline README.md]: this paragraph doesn't really make sense anymore, does it?

Comment (Sirupsen) [inline README.md]: I was a little too liberal with the updates to just `toxic` (mos people don't care about the distinction), thx

Comment (xthexder) [inline README.md]: Yeah, upstream latency is still possible, but it's not the default. We should definitely keep this example, but update the usage.

File: lib/toxiproxy/toxic_collection.rb [hunk 2]
```diff
@@ -13,34 +13,48 @@ def initialize(proxies)
...
 
     def apply(&block)
-      @toxics.each(&:enable)
-      yield
-    ensure
-      @toxics.each(&:disable)
+      names = toxics.group_by { |t| [t.name, t.proxy.name] }
+      dups  = names.values.select { |toxics| toxics.length > 1 }
+      if !dups.empty?
+        raise ArgumentError, "There are two toxics with the name #{dups.first[0]} for proxy #{dups.first[1]}, please override the default name (<type>_<direction>)"
```
Comment (xthexder) [inline lib/toxiproxy/toxic_collection.rb]: The default name is actually just `type`, unless I'm reading this wrong and `type_direction` is a suggested name.

Comment (Sirupsen) [inline lib/toxiproxy/toxic_collection.rb]: See my comment in https://github.com/Shopify/toxiproxy/pull/83 that we might want to default to `{type}_{stream}`.

**Comment by xthexder** [comment]
Should be good to go with this now. Apparently the update stopped supporting ruby 1.9 which wasn't ideal, so I went and changed it up.

**Pull request merged**